### ISSUE TITLE
Add recovery code

### ISSUE THREAD

**Issue by dginhoux** [authelia/authelia#1319]
Hi all, 

I think generate recovery code in case of second factor lost can be great...

Like Google, Microsoft and more.

**Comment by superadmin84** [comment]
I second this request. If this software is ever to be used by anyone other than the system administrator, it is important that users have account recovery options.

**Comment by james-d-elliott** [comment]
We will probably be figuring this out after a fairly drastic change to the verification process. What we're thinking of doing is removing the email links and utilizing a single use code system. The code would be valid to be entered for a period (probably 5 mins) and would allow the users the ability to setup 2FA devices for a total of 5 minutes.

We have to think about the method by which this is implemented and actually come up with a sound design as per the labels. 

I'm currently thinking the ability to generate the single use code email would be possible by someone who's 1FA'd. Then we can theoretically in the future implement this feature by giving administrators the configuration option to require 2FA if the user has a 2FA method configured. Then when users setup any device if they have no valid single use recovery codes generated we'd generate and display them to the user once they had successfully confirmed their 2FA device.

These single use recovery codes could then be utilized to bypass the 2FA requirement for the above rule. i.e. they would not count as 2FA for authentication, but would for the purposes of adding new devices.

**Comment by james-d-elliott** [comment]
This will be rolled up into #3801 where we'll overhaul areas surrounding this. We'll reopen this issue it doesn't end up being part of the final design.

**Comment by sphakka** [comment]
> This will be rolled up into #3801 where we'll overhaul areas surrounding this. We'll reopen this issue it doesn't end up being part of the final design.

Did this _recovery codes_ feature make it into #3801? Possibly missing something, but I can't find any related option in Authelia v4.38.7.

**Comment by Crowley723** [comment]
> 
> Did this _recovery codes_ feature make it into [#3801](https://github.com/authelia/authelia/issues/3801)? Possibly missing something, but I can't find any related option in Authelia v4.38.7.

No, recovery codes did not make it in. I'll go ahead and reopen this issue since its something we probably want.
